### PR TITLE
Jk 41

### PR DESCRIPTION
**Pull request by YugBandhara**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -40,6 +42,22 @@ const App = () => {
               <Route
                 path="/product/:id"
                 element={
+                  // <Suspense
```
Comment (sagun03) [inline src/App.jsx]: remove this cooment

File: node-shopping-server.code-workspace [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (sagun03) [inline node-shopping-server.code-workspace:1]: remove this sd this in gitignore

File: src/components/ProductRangeCard.jsx [hunk 3]
```diff
@@ -1,32 +1,42 @@
...
-    );
-    // TODO: snackbar issue
-    // setOpenAlert(true);
+    console.log(name, id, size, price, images)
+
+    const productObject = {
+      userId: user?.uid,
+      Products: [{
+        productID: id,
+        Quantity: 1,
```
Comment (sagun03) [inline src/components/ProductRangeCard.jsx]: why Quantity not quantity?
pplease use naming convention

File: src/components/ProductRangeCard.jsx [hunk 4]
```diff
@@ -1,32 +1,42 @@
...
+        unitPrice: price,
+        size: size
+      }]
+    }
+    const onSuccessCallback = () => {
+      window.location.reload();
```
Comment (sagun03) [inline src/components/ProductRangeCard.jsx:36]: why reloading the page??

File: src/hooks/useCart.jsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    },
+    onError: (error) => {
+      console.error("Error deleting cart:", error);
+    },
+    onSuccess: (_, { onSuccessCallback }) => {
+      console.log("Cart Deleted successfully");
+      // if (onSuccessCallback) {
+      //   onSuccessCallback();
+      // }
+      window.location.reload();
```
Comment (sagun03) [inline src/hooks/useCart.jsx]: why reloading?

File: src/hooks/useCart.jsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
```
Comment (sagun03) [inline src/hooks/useCart.jsx]: why reloading

File: src/pages/Cart.jsx [hunk 7]
```diff
@@ -246,25 +312,25 @@ const Cart = () => {
...
                                 onClick={() =>
-                                  handleClick("add", item, item.productId)
+                                  handleClick("add", item, item.productDetails?.id)
                                 }
                               />
                             </IconButton>
                           </ProductAmountContainer>
                           <ProductPrice>
-                            Rs. {item?.originalPrice * item?.quantity}
+                            Rs. {item?.productDetails?.sizes[0]?.price * item?.quantity}
```
Comment (sagun03) [inline src/pages/Cart.jsx:329]: why only sizes[0] what about other sizes object?

File: src/pages/Home.jsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
```
Comment (sagun03) [inline src/pages/Home.jsx]: remove this compoent

File: src/services/cartService.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    const response = await apiClient.request({
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      url: `/cart/${CartID}`,
+      data: {
+        userId: userId.toString(),
+        Products: Products
+      }
```
Comment (sagun03) [inline src/services/cartService.js]: can't we optimise it, it beig used in every function, too much redunant thing can we move this inside api.Client and just paas the method nd data from here?

File: src/hooks/useCart.jsx (hunk 6 above)
Comment (sagun03) [reply src/hooks/useCart.jsx]: invalidateQuery

**Review by sagun03 — CHANGES_REQUESTED** [review]

File: src/pages/Cart.jsx (hunk 7 above)
Comment (YugBandhara) [reply src/pages/Cart.jsx:329]: This is beacuse I am extracting the aztudal size like 250 ml por 500ml which is added to cart , and I am only sending one object in array of selected size

**Review by sagun03 — APPROVED** [review]

**Pull request merged**